tests/41end-to-end-keys/01-upload-key.pl

```diff
@@ -91,6 +91,12 @@
             device_keys => {
                user_id => "\@50-e2e-alice:localhost:8480",
                device_id => "alices_first_device",
+               algorithms => ["m.olm.curve25519-aes-sha256", "m.megolm.v1.aes-sha"],
+               keys => {
+                 "curve25519:".$user->device_id => "curve25519+key",
+                 "ed25519:".$user->device_id => "ed25519+key",
+               },
+               signatures => {},
             },
```

Comment:
This is caused by changes from https://github.com/element-hq/synapse/pull/17097 / https://github.com/element-hq/synapse/pull/19023 ?

I'm a bit confused about how Sytest is passing on Synapse `develop` if it's from those PRs.

Or since it's optional, why it's failing in (unknown, as described in PR title)?